woocommerce/woocommerce-ios
# Networking/Yosemite/UI: change how we save a Product

Fixes #1501 

## Changes

Note: this is a naive implementation of saving a Product as we discussed at the hangout yesterday, totally feel free to suggest anything else!
- Yosemite:
  - Created a `ProductUpdater` protocol that would return an immutable `Product` with certain fields updated
  - Added a `Product` extension to implement this protocol
  - Replaced the `Product` store/action to update Product name/description with the whole `Product`
- Networking:
  - Made `Product` implement `Codable`, and implemented `encode` for name and description properties (we will keep adding when we implement more edit features with tests)
  - Replaced `ProductsRemote`'s Product name/description functions with the whole Product
- App:
  - In `ProductFormViewController`, added a right navigation bar item "Done" (temporary action) to perform the actual Product save. When the user finishes updating Product description, we dismiss the editor and call `ProductUpdater`'s `descriptionUpdated` to create a new `Product` with the description updated

## Testing

- Go to Products tab
- Tap on a Product
- Tap "Edit"
- Tap on the Product description row and edit the description
- Tap "Done" in the editor --> on the Product main screen, the description should be updated locally but not remotely
- Tap "Done" on the main screen --> the Product should be updated remotely after the API request completes

Update release notes:
